traffic_ops/app/lib/API/Deliveryservice.pm

```diff
@@ -442,6 +442,139 @@ sub update {
+					"displayName"              => $rs->display_name,
+					"dnsBypassCname"           => $rs->dns_bypass_cname,
+					"dnsBypassIp"              => $rs->dns_bypass_ip,
+					"dnsBypassIp6"             => $rs->dns_bypass_ip6,
+					"dnsBypassTtl"             => $rs->dns_bypass_ttl,
+					"dscp"                     => $rs->dscp,
+					"edgeHeaderRewrite"        => $rs->edge_header_rewrite,
+					"exampleURLs"              => \@example_urls,
+					"geoLimitRedirectURL"      => $rs->geolimit_redirect_url,
+					"geoLimit"                 => $rs->geo_limit,
+					"geoLimitCountries"        => $rs->geo_limit_countries,
+					"geoProvider"              => $rs->geo_provider,
+					"globalMaxMbps"            => $rs->global_max_mbps,
+					"globalMaxTps"             => $rs->global_max_tps,
+					"httpBypassFqdn"           => $rs->http_bypass_fqdn,
+					"id"                       => $rs->id,
+					"infoUrl"                  => $rs->info_url,
+					"initialDispersion"        => $rs->initial_dispersion,
+					"ipv6RoutingEnabled"       => $rs->ipv6_routing_enabled,
+					"lastUpdated"              => $rs->last_updated,
+					"logsEnabled"              => $rs->logs_enabled,
+					"longDesc"                 => $rs->long_desc,
+					"longDesc1"                => $rs->long_desc_1,
+					"longDesc2"                => $rs->long_desc_2,
+					"matchList"                => \@matchlist,
+					"maxDnsAnswers"            => $rs->max_dns_answers,
+					"midHeaderRewrite"         => $rs->mid_header_rewrite,
+					"missLat"                  => defined($rs->miss_lat) ? 0.0 + $rs->miss_lat : undef,
+					"missLong"                 => defined($rs->miss_long) ? 0.0 + $rs->miss_long : undef,
+					"multiSiteOrigin"          => $rs->multi_site_origin,
+					"orgServerFqdn"            => $rs->org_server_fqdn,
+					"originShield"             => $rs->origin_shield,
+					"profileId"                => defined($rs->profile) ? $rs->profile->id : undef,
+					"profileName"              => defined($rs->profile) ? $rs->profile->name : undef,
+					"profileDescription"       => defined($rs->profile) ? $rs->profile->description : undef,
+					"protocol"                 => $rs->protocol,
+					"qstringIgnore"            => $rs->qstring_ignore,
+					"rangeRequestHandling"     => $rs->range_request_handling,
+					"regexRemap"               => $rs->regex_remap,
+					"regionalGeoBlocking"      => $rs->regional_geo_blocking,
+					"remapText"                => $rs->remap_text,
+					"signed"                   => $rs->signed,
+					"sslKeyVersion"            => $rs->ssl_key_version,
+					"trRequestHeaders"         => $rs->tr_request_headers,
+					"trResponseHeaders"        => $rs->tr_response_headers,
+					"type"                     => $rs->type->name,
+					"typeId"                   => $rs->type->id,
+					"xmlId"                    => $rs->xml_id
+				}
+			);
+
+			&log( $self, "Updated deliveryservice [ '" . $rs->xml_id . "' ] with id: " . $rs->id, "APICHANGE" );
```

Comment:
can you make this log statement a little more specific to a safe update. Maybe something like "Safe update performed for delivery service [ xmlId ] with id: 23"